tests/unit/src/main/scala/tests/TestingClient.scala

```diff
@@ -342,8 +343,21 @@ final class TestingClient(workspace: AbsolutePath, buffers: Buffers)
       .mkString("----\n")
   }
 
-  def applyCodeAction(codeAction: CodeAction): Unit = {
-    applyWorkspaceEdit(codeAction.getEdit())
+  private def executeServerCommand(
+      command: Command,
+      server: TestingServer
+  ): Unit = {
+    server.executeCommand(
+      command.getCommand(),
+      command.getArguments().asScala: _*
+    )
+  }
+
+  def applyCodeAction(codeAction: CodeAction, server: TestingServer): Unit = {
+    val edit = codeAction.getEdit()
+    val command = codeAction.getCommand()
+    if (edit != null) applyWorkspaceEdit(edit)
+    if (command != null) executeServerCommand(command, server)
```

Comment:
This is now as per the LSP specification: if `edit` and `command` are both present, first `edit` is applied then `command` is executed.